### PR TITLE
Document SSH user certificate authentication

### LINKED ISSUE #528: Connect using SSH certificates
With [^1] resolved the documentation must be complemented with instructions how to connect using SSH certificates instead of SSH keys.

> OpenSSH however also allows the use of SSH certificates to authenticate a server or a client, with the benefit of not requiring the prior storing of the key: an authority generates certificates that include the signature of the corresponding key, and the key used for the authentication of the server or client is trusted if the signature matches.

- [ ] https://docs.cyberduck.io/protocols/sftp/#private-key-files
- [ ] https://docs.cyberduck.io/protocols/sftp/#connecting
- [ ] https://docs.cyberduck.io/cyberduck/bookmarks/#bookmark-options

[^1]: iterate-ch/cyberduck#15183

### PR DESCRIPTION
**Pull request by AliveDevil**
Resolves #528

### CODE REVIEW

File: protocols/sftp/index.md [hunk 1]
```diff
@@ -96,7 +96,19 @@ PuTTY private keys (`.ppk`) are supported for `rsa` key types. `ed25519` is not
 
 #### OpenSSH Key Format Interoperability
 
-OpenSSH private keys of type `rsa`, `dsa`, `ecdsa` and `ed25519` (in OpenSSL`PEM` format) are supported. The new OpenSSH format (`openssh-key-v1`) is only supported for `ecdsa` and `ed25519`.
+OpenSSH private keys of type `rsa`, `dsa`, `ecdsa` and `ed25519` (in OpenSSL `PEM` format) are supported. The new OpenSSH format (`openssh-key-v1`) is only supported for `ecdsa` and `ed25519`.
+
+#### OpenSSH User Certificate Authentication
+
+:::{note}
+Applies to Cyberduck 8.9 and Mountain Duck 4.16
```
Comment (dkocher) [inline protocols/sftp/index.md:104]:
```suggestion
```{important}
* Cyberduck [8.9.0](https://cyberduck.io/changelog/) or later required
* Mountain Duck [4.16.0](https://mountainduck.io/changelog/) or later required
```
```

**Review by dkocher — CHANGES_REQUESTED** [review]

File: protocols/sftp/index.md [hunk 2]
```diff
@@ -96,7 +96,19 @@ PuTTY private keys (`.ppk`) are supported for `rsa` key types. `ed25519` is not
...
+
+Applies to SSH servers, which are configured with `TrustedUserCAKeys`, refer to your software vendor for configuration.
+:::
+
+Authentication using User CA signed private keys is supported, as long as both files (the private key, and the certificate file) are present at the same location.
+The certificate file has to be suffixed by either "-cert.pub" or ".pub" to be eligible for authentication. The OpenSSH configuration directive `CertificateFile` is not used.
```
[the remark this answers is not in the document]
Comment (AliveDevil) [reply protocols/sftp/index.md:110]: Not sure whether this needs a comment, as with any other public key authentication, the private key file has to be selected.

File: protocols/sftp/index.md [hunk 3]
```diff
@@ -96,7 +96,19 @@ PuTTY private keys (`.ppk`) are supported for `rsa` key types. `ed25519` is not
...
+Pay attention to the server configuration and `PubkeyAcceptedAlgorithms`, which private key algorithms are allowed. The algorithm used is `${Private Key Type}-cert-v01@openssh.com`.
```
[the remark this answers is not in the document]
Comment (AliveDevil) [reply protocols/sftp/index.md]: Removed the last sentence. Added all possible (?) values to Signatures.

File: protocols/sftp/index.md [hunk 4]
```diff
@@ -96,7 +96,21 @@ PuTTY private keys (`.ppk`) are supported for `rsa` key types. `ed25519` is not
...
+* Cyberduck [8.9.0](https://cyberduck.io/changelog/) or later required
+* Mountain Duck [4.16.0](https://mountainduck.io/changelog/) or later required
+
+Applies to SSH servers, which are configured with `TrustedUserCAKeys`, refer to your software vendor for configuration.
+:::
+
+Authentication using User CA signed private keys is supported, as long as both files (the private key, and the certificate file) are present at the same location.
+The certificate file has to be suffixed by either "-cert.pub" or ".pub" to be eligible for authentication. The OpenSSH configuration directive `CertificateFile` is not used.
+
+Pay attention to the server configuration and `PubkeyAcceptedAlgorithms`, which private key algorithms are allowed
```
Comment (dkocher) [inline protocols/sftp/index.md:113]:
```
Pay attention to the server configuration and `PubkeyAcceptedAlgorithms` specifically which determines the allowed private key algorithms to authenticate with.
```

File: protocols/sftp/index.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (dkocher) [reply protocols/sftp/index.md:111]: It must be made clear which file has to be selected.

File: protocols/sftp/index.md (hunk 4 above)
Comment (dkocher) [inline protocols/sftp/index.md:113]: Should we link to the OpenSSH documentation for the configuration directives?

File: protocols/sftp/index.md [hunk 5]
```diff
@@ -96,7 +96,21 @@ PuTTY private keys (`.ppk`) are supported for `rsa` key types. `ed25519` is not
...
-OpenSSH private keys of type `rsa`, `dsa`, `ecdsa` and `ed25519` (in OpenSSL`PEM` format) are supported. The new OpenSSH format (`openssh-key-v1`) is only supported for `ecdsa` and `ed25519`.
+OpenSSH private keys of type `rsa`, `dsa`, `ecdsa` and `ed25519` (in OpenSSL `PEM` format) are supported. The new OpenSSH format (`openssh-key-v1`) is only supported for `ecdsa` and `ed25519`.
+
+#### OpenSSH User Certificate Authentication
+
+:::{important}
+* Cyberduck [8.9.0](https://cyberduck.io/changelog/) or later required
+* Mountain Duck [4.16.0](https://mountainduck.io/changelog/) or later required
+
+Applies to SSH servers, which are configured with `TrustedUserCAKeys`, refer to your software vendor for configuration.
```
Comment (dkocher) [inline protocols/sftp/index.md:107]: Should we link to the OpenSSH documentation for the configuration directives?

File: protocols/sftp/index.md [hunk 6]
```diff
@@ -331,15 +340,23 @@ You can set Cyberduck or a third-party application as the default application (p
 
 ### Signatures
 
-`ssh-rsa`, `ssh-dss`, `ecdsa-sha2-nistp256`, `ecdsa-sha2-nistp384`, `ecdsa-sha2-nistp521`, `ssh-ed25519`
+`ssh-rsa`, `ssh-dss`, `ecdsa-sha2-nistp256`, `ecdsa-sha2-nistp384`, `ecdsa-sha2-nistp521`, `ssh-ed25519`, `rsa-sha2-256`, `rsa-sha2-512`
+
+With Cyberduck 8.9 and Mountain Duck 4.16, support for
```
Comment (dkocher) [inline protocols/sftp/index.md:345]: Drop the version requirement already clarified in note above.

File: protocols/sftp/index.md [hunk 7]
```diff
@@ -331,15 +340,23 @@ You can set Cyberduck or a third-party application as the default application (p
...
+- `ssh-rsa-cert-v01@openssh.com`
+- `ssh-dss-cert-v01@openssh.com`
+- `ecdsa-sha2-nistp256-cert-v01@openssh.com`
+- `ecdsa-sha2-nistp384-cert-v01@openssh.com`
+- `ecdsa-sha2-nistp521-cert-v01@openssh.com`
+- `ssh-ed25519-cert-v01@openssh.com`
```
Comment (dkocher) [inline protocols/sftp/index.md:351]: Split into sections for _SSH Key Types_  and _SSH Certificates Key Types_.

**Review by dkocher — CHANGES_REQUESTED** [review]

**Review by dkocher — APPROVED** [review]

**Pull request merged**